### ISSUE TITLE
torch.bincount beyond 1d array

### ISSUE THREAD

**Issue by interesaaat** [pytorch/pytorch#43867]
## 🚀 Feature
Currently `torch.bincount` only works over 1d arrays. It would be good to have it working also on tensors of arbitrary shape. This feature for example is available in Tensorflow, where you can specify the dimension on which to apply `bincount`.

## Motivation

In [Hummingbird](https://github.com/microsoft/hummingbird/) we are using `torch.mode` for multiclass SVM. However (1) `torch.mode` does not work on GPU and (2) TVM does not support mode but they are willing to add bincount. More in [this] (https://github.com/microsoft/hummingbird/issues/232#issuecomment-682284632) discussion.

## Pitch

Having `torc.bincount` accepting a dim argument where you can specify on which dimension to apply bincount.

**Comment by vadimkantorov** [comment]
Also stumbled on this
